pkg/controllers/dnspolicy/dns_helper_test.go

```diff
@@ -16,192 +18,159 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
-
-	"github.com/Kuadrant/multicluster-gateway-controller/pkg/apis/v1alpha1"
-	"github.com/Kuadrant/multicluster-gateway-controller/pkg/dns"
-	"github.com/Kuadrant/multicluster-gateway-controller/pkg/traffic"
-	testutil "github.com/Kuadrant/multicluster-gateway-controller/test/util"
 )
 
-func TestDNS_GetDNSRecords(t *testing.T) {
-	cases := []struct {
-		Name      string
-		MZ        func() *v1alpha1.ManagedZone
-		SubDomain string
-		Assert    func(t *testing.T, dnsRecord *v1alpha1.DNSRecord, err error)
-		DNSRecord *v1alpha1.DNSRecord
-		Gateway   *gatewayv1beta1.Gateway
-	}{
-		{
-			Name: "test get dns record returns record",
-			MZ: func() *v1alpha1.ManagedZone {
-				return &v1alpha1.ManagedZone{
-					ObjectMeta: v1.ObjectMeta{
-						Name:      "b.c.com",
-						Namespace: "test",
-					},
-					Spec: v1alpha1.ManagedZoneSpec{
-						DomainName: "b.c.com",
-					},
-				}
-			},
-			SubDomain: "a",
-			DNSRecord: &v1alpha1.DNSRecord{
-				ObjectMeta: v1.ObjectMeta{
-					Name:      "a.b.c.com",
-					Namespace: "test",
-					Labels: map[string]string{
-						dns.LabelGatewayReference: "test",
-					},
-				},
-			},
-			Gateway: &gatewayv1beta1.Gateway{
-				ObjectMeta: v1.ObjectMeta{
-					UID: types.UID("test"),
-				},
-			},
+func testScheme(t *testing.T) *runtime.Scheme {
+	scheme := runtime.NewScheme()
+	if err := v1alpha1.AddToScheme(scheme); err != nil {
+		t.Fatalf("falied to add work scheme %s ", err)
+	}
+	if err := gatewayv1beta1.AddToScheme(scheme); err != nil {
+		t.Fatalf("falied to add work scheme %s ", err)
+	}
+	return scheme
+}
 
-			Assert: func(t *testing.T, dnsRecord *v1alpha1.DNSRecord, err error) {
-				if err != nil {
-					t.Fatalf("expectd no error but got %s", err)
-				}
-			},
+func TestSetProviderSpecific(t *testing.T) {
+	endpoint := &v1alpha1.Endpoint{
+		ProviderSpecific: []v1alpha1.ProviderSpecificProperty{
+			{Name: "weight", Value: "120"},
 		},
+	}
+
+	// Test updating an existing property
+	endpoint.SetProviderSpecific("weight", "60")
+	for _, property := range endpoint.ProviderSpecific {
+		if property.Name == "weight" && property.Value != "60" {
+			t.Errorf("Existing property was not updated. Got %s, expected 60", property.Value)
+		}
+	}
+}
+
+func Test_CreateDNSRecord(t *testing.T) {
```

Comment:
nit: most of the tests in here still don't match what would be expected if you generated them. I think this would normally be  `Test_dnsHelper_createDNSRecord` to match the name of the receiver and method since it's private.